core/src/data_sources/data_source.rs

```diff
@@ -790,33 +826,29 @@ impl DataSource {
                 match f.tags {
                     Some(tags) => {
                         match tags.is_in.clone() {
-                            Some(v) => must_filter.push(
-                                qdrant::FieldCondition {
-                                    key: "tags".to_string(),
-                                    r#match: Some(qdrant::Match {
-                                        match_value: Some(qdrant::r#match::MatchValue::Keywords(
-                                            qdrant::RepeatedStrings { strings: v },
-                                        )),
-                                    }),
-                                    ..Default::default()
-                                }
-                                .into(),
-                            ),
+                            Some(v) => must_filter.push(qdrant_match_field_condition("tags", v)),
                             None => (),
                         };
                         match tags.is_not.clone() {
-                            Some(v) => must_not_filter.push(
-                                qdrant::FieldCondition {
-                                    key: "tags".to_string(),
-                                    r#match: Some(qdrant::Match {
-                                        match_value: Some(qdrant::r#match::MatchValue::Keywords(
-                                            qdrant::RepeatedStrings { strings: v },
-                                        )),
-                                    }),
-                                    ..Default::default()
-                                }
-                                .into(),
-                            ),
+                            Some(v) => {
+                                must_not_filter.push(qdrant_match_field_condition("tags", v))
+                            }
+                            None => (),
+                        };
+                    }
+                    None => (),
+                };
+
+                match f.parents {
+                    Some(tags) => {
```

Comment:
let's not use `tags` as variable name here

Reply:
:fearful: